### ISSUE TITLE
loss = nan

### ISSUE THREAD

**Issue by Mobzhang** [rootlu/MMDNE#3]
When I use my datasets,some of them can be train successfully, some will get a problem loss = nan. did anyone have same issue with me? and why?

**Comment by FAME-code** [comment]
I also encountered the same problem, reducing the learning rate can alleviate this problem, but not always work.

**Comment by Mobzhang** [comment]
> I also encountered the same problem, reducing the learning rate can alleviate this problem, but not always work.



> I also encountered the same problem, reducing the learning rate can alleviate this problem, but not always work.
I clip the datasets and reduce the batchsize,  and I notice it can work at first. Does it mean my dataset has dirty data ? but my data could use in other dynamic method.

**Comment by rootlu** [comment]
Hi, 

Due to the joint modeling of micro- and macro-dynamics, the optimization of the model needs to be careful. Maybe you can try to tune the $\epsilon$ in the loss function and the learning rate. 
Meanwhile, please make sure that the data format is correct.

Thanks for your interest in our work.

**Comment by Mobzhang** [comment]
thanks for author replying. while i tune the epsilon and adjust learning rate to 0.0001， it still doesnt work. Whether the model will be affected, if the graph has some node which has low degree (1, 2, ... , )? I use HTNE to train this graph, it can train successfully. I think the data format in HTNE and MMDNE  is same, So I still dont know why？ -。-

**Comment by rootlu** [comment]
Hi, 

The low degree may affect the model,  since the calculations of attention and macro-dynamics are related to the graph scale. But I haven't verified this point with experiments.
